diff --git a/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js b/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js
index 56ca29a..f3bb142 100644
--- a/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js
+++ b/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js
@@ -30,7 +30,7 @@ module.exports = {
 
               const onlyProperty = test.properties.find(
                 property =>
-                  property.key.type === 'Identifier' &&
+                  property.key?.type === 'Identifier' &&
                   property.key.name === 'only' &&
                   property.value.type === 'Literal' &&
                   property.value.value
diff --git a/node_modules/eslint-plugin-eslint-plugin/lib/rules/require-meta-docs-url.js b/node_modules/eslint-plugin-eslint-plugin/lib/rules/require-meta-docs-url.js
index ed7d67a..af8ea5c 100644
--- a/node_modules/eslint-plugin-eslint-plugin/lib/rules/require-meta-docs-url.js
+++ b/node_modules/eslint-plugin-eslint-plugin/lib/rules/require-meta-docs-url.js
@@ -48,7 +48,7 @@ module.exports = {
     const options = context.options[0] || {};
     const sourceCode = context.getSourceCode();
     const filename = context.getFilename();
-    const ruleName = filename === '<input>' ? undefined : path.basename(filename, '.js');
+    const ruleName = filename === '<input>' ? undefined : path.basename(filename, '.ts');
     const expectedUrl = !options.pattern || !ruleName
       ? undefined
       : options.pattern.replace(/{{\s*name\s*}}/g, ruleName);
diff --git a/node_modules/eslint-plugin-eslint-plugin/lib/utils.js b/node_modules/eslint-plugin-eslint-plugin/lib/utils.js
index 3b27af1..9754663 100644
--- a/node_modules/eslint-plugin-eslint-plugin/lib/utils.js
+++ b/node_modules/eslint-plugin-eslint-plugin/lib/utils.js
@@ -1,6 +1,6 @@
 'use strict';
 
-const { getStaticValue } = require('eslint-utils');
+const { getStaticValue, findVariable } = require('eslint-utils');
 const estraverse = require('estraverse');
 
 /**
@@ -77,10 +77,10 @@ function isNormalFunctionExpressionReference (node, scopeManager) {
 */
 function isRuleTesterConstruction (node) {
   return node.type === 'NewExpression' && (
-    (node.callee.type === 'Identifier' && node.callee.name === 'RuleTester') ||
+    (node.callee.type === 'Identifier' && node.callee.name === 'GraphQLRuleTester') ||
     (node.callee.type === 'MemberExpression' &&
       node.callee.property.type === 'Identifier' &&
-      node.callee.property.name === 'RuleTester')
+      node.callee.property.name === 'GraphQLRuleTester')
   );
 }
 
@@ -138,12 +138,18 @@ function isFunctionRule (node) {
 /**
  * Helper for `getRuleInfo`. Handles ESM and TypeScript rules.
  */
-function getRuleExportsESM (ast) {
+function getRuleExportsESM (ast, scopeManager) {
   return ast.body
     .filter(statement => statement.type === 'ExportDefaultDeclaration')
     .map(statement => statement.declaration)
     // eslint-disable-next-line unicorn/prefer-object-from-entries
     .reduce((currentExports, node) => {
+      // Check `const rule = { create() {}, meta: {} }; export default rule`
+      if (node.type === 'Identifier' && scopeManager) {
+        const variable = findVariable(scopeManager.globalScope, node);
+        return collectInterestingProperties(variable.defs[0].node.init.properties, INTERESTING_RULE_KEYS);
+      }
+
       if (node.type === 'ObjectExpression') {
         // Check `export default { create() {}, meta: {} }`
         return collectInterestingProperties(node.properties, INTERESTING_RULE_KEYS);
@@ -244,7 +250,7 @@ module.exports = {
   from the file, the return value will be `null`.
   */
   getRuleInfo ({ ast, scopeManager }) {
-    const exportNodes = ast.sourceType === 'module' ? getRuleExportsESM(ast) : getRuleExportsCJS(ast);
+    const exportNodes = ast.sourceType === 'module' ? getRuleExportsESM(ast, scopeManager) : getRuleExportsCJS(ast);
 
     const createExists = Object.prototype.hasOwnProperty.call(exportNodes, 'create');
     if (!createExists) {
@@ -335,7 +341,7 @@ module.exports = {
           variableIdentifiers.has(statement.expression.callee.object)
         ) &&
         statement.expression.callee.property.type === 'Identifier' &&
-        statement.expression.callee.property.name === 'run'
+        statement.expression.callee.property.name === 'runGraphQLTests'
       ) {
         runCalls.push(statement.expression);
       }
