diff --git a/node_modules/eslint/lib/rule-tester/rule-tester.js b/node_modules/eslint/lib/rule-tester/rule-tester.js
index cac81bc..3a3dbbe 100644
--- a/node_modules/eslint/lib/rule-tester/rule-tester.js
+++ b/node_modules/eslint/lib/rule-tester/rule-tester.js
@@ -905,7 +905,17 @@ class RuleTester {
                         "Expected no autofixes to be suggested"
                     );
                 } else {
-                    assert.strictEqual(result.output, item.output, "Output is incorrect.");
+                    if (item.output.includes('# normalize graphql')) {
+                        const graphql = require('graphql');
+  
+                        function normalize(value) {
+                            return graphql.print(graphql.parse(value.replace('# normalize graphql', '')));
+                        }
+  
+                        assert.strictEqual(normalize(result.output), normalize(item.output), 'Output is incorrect.');
+                    } else {
+                        assert.strictEqual(result.output, item.output, 'Output is incorrect.');
+                    }
                 }
             } else {
                 assert.strictEqual(
