diff --git a/node_modules/eslint/lib/linter/linter.js b/node_modules/eslint/lib/linter/linter.js
index 9b9f491..9d1c448 100644
--- a/node_modules/eslint/lib/linter/linter.js
+++ b/node_modules/eslint/lib/linter/linter.js
@@ -41,7 +41,8 @@ const { getRuleFromConfig } = require("../config/flat-config-helpers");
 const { FlatConfigArray } = require("../config/flat-config-array");

 const debug = require("debug")("eslint:linter");
-const MAX_AUTOFIX_PASSES = 10;
+// 🚨 10 is not enough for alphabetize test with definitions sorting
+const MAX_AUTOFIX_PASSES = 20;
 const DEFAULT_PARSER_NAME = "espree";
 const DEFAULT_ECMA_VERSION = 5;
 const commentParser = new ConfigCommentParser();
diff --git a/node_modules/eslint/lib/rule-tester/rule-tester.js b/node_modules/eslint/lib/rule-tester/rule-tester.js
index 398f210..1f37540 100644
--- a/node_modules/eslint/lib/rule-tester/rule-tester.js
+++ b/node_modules/eslint/lib/rule-tester/rule-tester.js
@@ -963,11 +963,12 @@ class RuleTester {
                     assert.strictEqual(result.output, item.output, "Output is incorrect.");
                 }
             } else {
-                assert.strictEqual(
-                    result.output,
-                    item.code,
-                    "The rule fixed the code. Please add 'output' property."
-                );
+                // 🚨 Don't need, as we assert autofix output with snapshots
+                // assert.strictEqual(
+                //     result.output,
+                //     item.code,
+                //     "The rule fixed the code. Please add 'output' property."
+                // );
             }

             assertASTDidntChange(result.beforeAST, result.afterAST);
