diff --git a/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js b/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js
index caa646e..c3fb975 100644
--- a/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js
+++ b/node_modules/eslint-plugin-eslint-plugin/lib/rules/no-only-tests.js
@@ -82,7 +82,7 @@ module.exports = {
               test.type === 'CallExpression' &&
               test.callee.type === 'MemberExpression' &&
               test.callee.object.type === 'Identifier' &&
-              test.callee.object.name === 'RuleTester' &&
+              test.callee.object.name === 'GraphQLRuleTester' &&
               test.callee.property.type === 'Identifier' &&
               test.callee.property.name === 'only'
             ) {
diff --git a/node_modules/eslint-plugin-eslint-plugin/lib/utils.js b/node_modules/eslint-plugin-eslint-plugin/lib/utils.js
index 3129903..be85e4a 100644
--- a/node_modules/eslint-plugin-eslint-plugin/lib/utils.js
+++ b/node_modules/eslint-plugin-eslint-plugin/lib/utils.js
@@ -25,10 +25,10 @@ function isNormalFunctionExpression(node) {
 function isRuleTesterConstruction(node) {
   return (
     node.type === 'NewExpression' &&
-    ((node.callee.type === 'Identifier' && node.callee.name === 'RuleTester') ||
+    ((node.callee.type === 'Identifier' && node.callee.name === 'GraphQLRuleTester') ||
       (node.callee.type === 'MemberExpression' &&
         node.callee.property.type === 'Identifier' &&
-        node.callee.property.name === 'RuleTester'))
+        node.callee.property.name === 'GraphQLRuleTester'))
   );
 }
 
