diff --git a/package.json b/package.json
index 19e4fa3271a65f66a3712707a69686212a263509..39d2e0e71f0b97319cc497996de3d85c93a059b2 100644
--- a/package.json
+++ b/package.json
@@ -10,6 +10,7 @@
   "exports": {
     "./package.json": "./package.json",
     ".": "./lib/api.js",
+    "./linter": "./lib/linter/index.js",
     "./use-at-your-own-risk": "./lib/unsupported-api.js"
   },
   "scripts": {
diff --git a/lib/linter/linter.js b/lib/linter/linter.js
index a29ce92..a574871 100644
--- a/lib/linter/linter.js
+++ b/lib/linter/linter.js
@@ -41,7 +41,8 @@ const { getRuleFromConfig } = require("../config/flat-config-helpers");
 const { FlatConfigArray } = require("../config/flat-config-array");

 const debug = require("debug")("eslint:linter");
-const MAX_AUTOFIX_PASSES = 10;
+// 🚨 10 is not enough for alphabetize test with definitions sorting
+const MAX_AUTOFIX_PASSES = 20;
 const DEFAULT_PARSER_NAME = "espree";
 const DEFAULT_ECMA_VERSION = 5;
 const commentParser = new ConfigCommentParser();
diff --git a/lib/rule-tester/rule-tester.js b/lib/rule-tester/rule-tester.js
index 2af272b..738f80f 100644
--- a/lib/rule-tester/rule-tester.js
+++ b/lib/rule-tester/rule-tester.js
@@ -1005,11 +1005,12 @@ class RuleTester {
                     assert.strictEqual(result.output, item.output, "Output is incorrect.");
                 }
             } else {
-                assert.strictEqual(
-                    result.output,
-                    item.code,
-                    "The rule fixed the code. Please add 'output' property."
-                );
+                // 🚨 Don't need, as we assert autofix output with snapshots
+                // assert.strictEqual(
+                //     result.output,
+                //     item.code,
+                //     "The rule fixed the code. Please add 'output' property."
+                // );
             }

             assertASTDidntChange(result.beforeAST, result.afterAST);
