diff --git a/node_modules/eslint/lib/rule-tester/rule-tester.js b/node_modules/eslint/lib/rule-tester/rule-tester.js
index 398f210..2d4c0f8 100644
--- a/node_modules/eslint/lib/rule-tester/rule-tester.js
+++ b/node_modules/eslint/lib/rule-tester/rule-tester.js
@@ -960,14 +960,25 @@ class RuleTester {
                         "Expected no autofixes to be suggested"
                     );
                 } else {
-                    assert.strictEqual(result.output, item.output, "Output is incorrect.");
+                    if (item.output.includes('# normalize graphql')) {
+                        const graphql = require('graphql');
+
+                        function normalize(value) {
+                            return graphql.print(graphql.parse(value.replace('# normalize graphql', '')));
+                        }
+
+                        assert.strictEqual(normalize(result.output), normalize(item.output), 'Output is incorrect.');
+                    } else {
+                        assert.strictEqual(result.output, item.output, 'Output is incorrect.');
+                    }
                 }
             } else {
-                assert.strictEqual(
-                    result.output,
-                    item.code,
-                    "The rule fixed the code. Please add 'output' property."
-                );
+                // 🚨 Don't need, as we assert autofix output with snapshots
+                // assert.strictEqual(
+                //     result.output,
+                //     item.code,
+                //     "The rule fixed the code. Please add 'output' property."
+                // );
             }

             assertASTDidntChange(result.beforeAST, result.afterAST);
