diff --git a/node_modules/json-schema-to-markdown/index.js b/node_modules/json-schema-to-markdown/index.js
index 6b805f9..8734419 100644
--- a/node_modules/json-schema-to-markdown/index.js
+++ b/node_modules/json-schema-to-markdown/index.js
@@ -71,7 +71,8 @@ function generateSchemaSectionText(octothorpes, name, isRequired, schema, subSch
 			})
 		}
 	} else if (schemaType === 'array') {
-		var itemsType = schema.items && schema.items.type
+		schema.items = schema.items || {};
+		var itemsType = schema.items.type
 
 		if (!itemsType && schema.items['$ref']) {
 			itemsType = getActualType(schema.items, subSchemas)
@@ -96,6 +97,9 @@ function generateSchemaSectionText(octothorpes, name, isRequired, schema, subSch
 			} else if (schema.items.not) {
 				text.push('The elements of the array must *not* match the following properties:')
 				validationItems = schema.items.not
+			} else if (schema.contains?.enum) {
+				text.push('The elements of the array must contain the following properties:')
+				text.push(schema.contains.enum.map(enumItem => `- \`${enumItem}\``).join('\n'));
 			}
 
 			if (validationItems.length > 0) {
