diff --git a/dist/chunk-runtime-setup.f79addc3.js b/dist/chunk-runtime-setup.f79addc3.js
index 706897e9584170e82fe449d2edcc663247597d67..954833f0531db9d4f0f685b9429266b8099c5432 100644
--- a/dist/chunk-runtime-setup.f79addc3.js
+++ b/dist/chunk-runtime-setup.f79addc3.js
@@ -483,7 +483,7 @@ async function runSuite(suite) {
     if (!hasTests(suite)) {
       suite.result.state = "fail";
       if (!suite.result.error)
-        suite.result.error = new Error(`No test found in suite ${suite.name}`);
+        suite.result.state = "pass";
     } else if (hasFailed(suite)) {
       suite.result.state = "fail";
     } else {