# `no-anonymous-operations`

‚úÖ The `"extends": "plugin:@graphql-eslint/operations-recommended"` property in a configuration file
enables this rule.

üí° This rule provides
[suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions)

- Category: `Operations`
- Rule name: `@graphql-eslint/no-anonymous-operations`
- Requires GraphQL Schema: `false`
  [‚ÑπÔ∏è](/docs/getting-started#extended-linting-rules-with-graphql-schema)
- Requires GraphQL Operations: `false`
  [‚ÑπÔ∏è](/docs/getting-started#extended-linting-rules-with-siblings-operations)

Require name for your GraphQL operations. This is useful since most GraphQL client libraries are
using the operation name for caching purposes.

## Usage Examples

### Incorrect

```graphql
# eslint @graphql-eslint/no-anonymous-operations: 'error'

query {
  # ...
}
```

### Correct

```graphql
# eslint @graphql-eslint/no-anonymous-operations: 'error'

query user {
  # ...
}
```

## Resources

- [Rule source](https://github.com/B2o5T/graphql-eslint/tree/master/packages/plugin/src/rules/no-anonymous-operations.ts)
- [Test source](https://github.com/B2o5T/graphql-eslint/tree/master/packages/plugin/__tests__/no-anonymous-operations.spec.ts)
